modules/ui/sections/map_daterange.js

```diff
@@ -0,0 +1,167 @@
+import { dispatch as d3_dispatch } from 'd3-dispatch';
+import { select as d3_select } from 'd3-selection';
+
+import { t } from '../../core/localizer';
+import { uiTooltip } from '../tooltip';
+import { uiSection } from '../section';
+import { utilQsString, utilStringQs } from '../../util';
+
+const DEFAULT_MIN_DATE = '-4000-01-01';
+const DEFAULT_MAX_DATE = (new Date()).getFullYear() + '-12-31';
+
+const INPUT_STYLES = [
+    { name: 'width', value: '125px' },
+    { name: 'text-align', value: 'center' },
+];
+const LABEL_STYLES = [
+    { name: 'font-weight', value: 'bold' },
+    { name: 'display', value: 'inline-block' },
+    { name: 'width', value: '75px' },
+];
+
+export function uiSectionDateRange(context) {
+    // despite appearing as a separate panel, Map Features does the real filtering
+    // see applyDateRange() in this panel, where the dateRange value is set
+    // see modules/renderer/features.js checkDateFilter() which applies the filters
+    // see modules/renderer/features.js update() which updates URL params
+
+    const section = uiSection('date_ranges', context)
+        .label(t('date_ranges.title'))
+        .disclosureContent(renderDisclosureContent)
+        .expandedByDefault(false);
+
+    function renderDisclosureContent(selection) {
+        const container = selection.selectAll('.date_ranges-container').data([0]);
+
+        // start date label & input
+        const mindate_label = container.enter()
+            .append('label')
+            .html(t('date_ranges.start_date.description'))
+            .merge(container);
+        const mindate_input = container.enter()
+            .append('input')
+            .attr('type', 'text')
+            .attr('value', DEFAULT_MIN_DATE)
+            .attr('title', t('date_ranges.start_date.tooltip'))
+            .attr('placeholder', t('date_ranges.start_date.placeholder'))
+            .merge(container);
+
+        // line break
+        container.enter()
+            .append('br')
+            .merge(container);
+
+        // end date label & input
+        const maxdate_label = container.enter()
+            .append('label')
+            .html(t('date_ranges.end_date.description'))
+            .merge(container);
+        const maxdate_input = container.enter()  // we will refer to this widget by its name attribute to fetch our date range
+            .append('input')
+            .attr('type', 'text')
+            .attr('value', DEFAULT_MAX_DATE)
+            .attr('title', t('date_ranges.end_date.tooltip'))
+            .attr('placeholder', t('date_ranges.end_date.placeholder'))
```

Comment:
`t` was changed upstream to avoid cross-site scripting attacks. Now you need to call `t.append()`.

Reply:
Changed.